instrumentor/controllers/common.go

```diff
@@ -268,47 +274,57 @@ func processInstrumentedApps(ctx context.Context, podTemplateSpec *v1.PodTemplat
 			logger.Error(err, "Error getting object")
 			return err
 		}
+		err = c.Get(ctx, client.ObjectKeyFromObject(&instApp), &instApp)
+		if err != nil {
+			logger.Error(err, "Error getting instrumented application")
+			return err
+		}
 		logger.V(0).Info("Instrumenting pod: " + podTemplateSpec.GetName())
 		err = patch.ModifyObject(podTemplateSpec, &instApp)
 		if err != nil {
 			logger.Error(err, "error patching deployment / statefulset")
 			return err
 		}
-		// Define an exponential backoff configuration
-		backoff := wait.Backoff{
-			Duration: time.Second * 2, // Initial delay
-			Factor:   2,               // Factor by which the delay is exponentially increased
-			Jitter:   0.1,             // Jitter to introduce some random variation in the delay
-			Steps:    5,               // Number of steps to retry
-		}
-		// The error variable to collect all errors encountered
-		var lastErr error
-		// Retry logic with exponential backoff
-		retryErr := wait.ExponentialBackoff(backoff, func() (bool, error) {
-			updateErr := c.Update(ctx, object)
-			if updateErr != nil {
-				// Save the error encountered
-				lastErr = updateErr
-				logger.Error(updateErr, "error instrumenting application, retrying...")
-				// Return false to indicate a retry should happen
-				return false, nil
-			}
-			err := c.Status().Update(ctx, &instApp)
-			if err != nil {
-				logger.Error(err, "error updating instrumented status")
-				return false, nil
-			} else {
-				instApp.Status.TracesInstrumented = false
-			}
-			// Return true to indicate the function was successful
-			return true, nil
-		})
-
-		if retryErr != nil || lastErr != nil {
-			if retryErr != nil {
-				logger.Error(lastErr, "error after retrying")
-			}
-			return lastErr
+		err = c.Update(ctx, object)
+		if err != nil {
+			logger.Error(err, "error instrumenting application ")
+			return err
+		}
+		err = c.Update(ctx, &instApp)
+		if err != nil {
+			logger.Error(err, "error updating custom resource instrumented status")
+			return err
+		}
+		// instApp.Status.TracesInstrumented is a part of the status in the custom resource definition
+		instApp.Status.TracesInstrumented = true
+		err = c.Status().Update(ctx, &instApp)
+		if err != nil {
+			logger.Error(err, "error computing traces instrumented status")
+			return err
+		}
+
+	}
+	// if the app is instrumented update the active service name
+	if instrumented {
+		err = c.Get(ctx, client.ObjectKey{Namespace: object.GetNamespace(), Name: object.GetName()}, object)
+		if err != nil {
+			logger.Error(err, "Error getting object")
+			return err
+		}
+		err = patch.UpdateActiveServiceName(podTemplateSpec, &instApp)
+		if err != nil {
+			logger.Error(err, "error updating active service name for deployment / statefulset")
```

Comment:
It'll not be fun to remember updating this line whenever we'll support a new resource. Please change it to a generic term 🙏